### PR TITLE
Auth toast

### PR DESCRIPTION
**Pull request by rehanmohiuddin**
- Added localstorage to store token
- Added get user when app opens from localstorage
- Added Log out
- Added Custom Header Auth Actions
- Revamped UI

Files To Check : 
components folder:
- Toast
- Auth

context/reducers/actions folder :
- toast
- auth

### CODE REVIEW

**Review by Richa-rani02 — APPROVED** [review]
Code is clean and easily understandable
liked that you have handled errors properly

File: src/Components/AuthLeft/index.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import Logo from "../../assets/SawaalJawab -mailer.png";
+import { BUTTON } from "../../Util/constants";
+import Button from "../Button";
+
+function Index() {
+  return (
+    <div className="login-left">
+      <img src={Logo} />
```
Comment (shivanigangadharan) [inline src/Components/AuthLeft/index.jsx:9]: can provide an alt message here

File: src/Components/Header/index.jsx [hunk 2]
```diff
@@ -1,54 +1,110 @@
...
+
   const options = [
+    // {
+    //   name: "Home",
+    //   route: "/",
+    // },
+    // {
+    //   name: "Create Quiz",
+    //   route: "/create/quiz",
+    // },
```
Comment (shivanigangadharan) [inline src/Components/Header/index.jsx:42]: can remove comments if not reqd

File: src/Components/Header/index.jsx [hunk 3]
```diff
@@ -1,54 +1,110 @@
...
+    </>
   );
+
   return (
     <header>
-      <div>
+      <div className="header-container">
         <Link to={"/"}>
-          <img className="logo" src={Logo} />
+          <img src={Logo} />
```
Comment (shivanigangadharan) [inline src/Components/Header/index.jsx:95]: alt here too

File: src/Components/Header/index.scss [hunk 4]
```diff
@@ -4,26 +4,81 @@
...
+  }
+  border-bottom: 1px solid $pale-gray;
+  box-shadow: 0 0.1rem 0.1rem 0 $pale-gray;
+  img {
+    width: calc($spacing-unit * 11);
+  }
+  .home-header {
+    div {
+      font-family: Poppins, sans-serif;
+      font-size: calc($spacing-unit * 1.2);
```
Comment (shivanigangadharan) [inline src/Components/Header/index.scss:30]: I like how you've used calc function in these places

**Review by shivanigangadharan — APPROVED** [review]
Code looks really good, got to learn from it as well.

**Pull request merged**